### PR TITLE
[i18n] Save preferred_language for auth user in db

### PR DESCRIPTION
**Pull request by rajatvijay**
Relates #2927 

**Technical details**

This PR does the following things - 

1. Adds `Locale` middleware in BE
2. Add a `preferred_language` attribute in the `users` model and serializers. Migrations for this field.
3. Adds the `Preferred Language` dropdown in the user's profile page. 
4. Load the changed preferred language's translations from the profile page dynamically, on the client side. 

**Screenshots**

https://github.com/centerofci/mathesar/assets/11032856/0ac0cb2f-c699-4d99-8c79-011cd79da681

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by rajatvijay** [comment]
@ghislaineguerin @kgodey assigned you for a product related review.

**Comment by ghislaineguerin** [comment]
@rajatvijay Looks good to me.

**Comment by seancolsen** [comment]
"Preferred Language" and "Display Language" both seem fine to me. But "UI Language" seems like is has the potential to confuse some people who are not familiar with that acronym.

**Comment by kgodey** [comment]
Let's do "Display Language" then.

**Review by silentninja — APPROVED** [review]
I reviewed the changes to the backend code and it looks good to me!

File: mathesar_ui/src/api/users.ts
Comment (seancolsen) [inline mathesar_ui/src/api/users.ts:15]: Can we type this as `Locales` instead of `string`? I haven't looked deeply, but it seems like that might eliminate the need for the type assertion `as Locales` that you have in `UserDetailsForm.svelte`, and it would maybe allow us to also use `Locales` instead of `string` in `SelectPreferredLanguage.svelte`

**Review by seancolsen — CHANGES_REQUESTED** [review]
I've noted a couple small comments on the diff, but a larger issue here is that we'll need to find a way to hide this UI before we can merge it. Otherwise we should hold off on merging until we have the full Japanese translation available.

File: mathesar_ui/src/systems/users-and-permissions/UserDetailsForm.svelte
Comment (seancolsen) [inline mathesar_ui/src/systems/users-and-permissions/UserDetailsForm.svelte:156]: Also "Preferred Language" needs to be changed to "Display Language" here.

Comment (rajatvijay) [reply mathesar_ui/src/systems/users-and-permissions/UserDetailsForm.svelte]: 0e8e674

File: mathesar_ui/src/api/users.ts
Comment (rajatvijay) [reply mathesar_ui/src/api/users.ts]: 75a5e4b

File: mathesar_ui/src/systems/users-and-permissions/SelectPreferredLanguage.svelte
Comment (rajatvijay) [reply mathesar_ui/src/systems/users-and-permissions/SelectPreferredLanguage.svelte]: 79d50b4

**Comment by rajatvijay** [comment]
Commented out the `Display Language` select on the profile page 36d314d

**Review by seancolsen — APPROVED** [review]
Looks good now 👍🏻

**Comment by seancolsen** [comment]
I'm holding off merge since this is stacked on top of #3102. @rajatvijay feel free to merge this yourself once that PR is merged.

**Comment by seancolsen** [comment]
I figured it out &mdash; it was an issue with accepting user input when using `docker exec`. I was able to generate a new migration by running:

```
sudo docker exec mathesar_service_dev sh -c "python manage.py makemigrations --merge --no-input"
```

I committed that migration in 825a36f7f15be5650c550b9bd395a9441421b953

**Comment by seancolsen** [comment]
If all the tests pass, I'm assuming we're good to go here, so I've queued this to merge now.

**Pull request merged**
